### DISCUSSION TITLE
Method to check if a window is open or more general info about a window

### DISCUSSION THREAD

**Discussion by martinpl** [NativePHP/laravel#247] [Ideas]
There is no option to check if window is opened. Maybe folks need more information about window so this is kind open for discussion. I need this to toggle (close / open) window.

**Comment by simonhamp** [comment]
It would be good to get a better idea of what you're trying to achieve, but on the surface I think you're right that this is missing.

Because of the nature of PHP's request cycle and overall statelessness, it's 'non-trivial' to keep tabs on the windows available to your application.

Not saying it's 'hard', necessarily; it could be as 'simple' as a bit of caching so that Window objects and config are available across multiple requests. Or it could be a request back over to Electron to fetch window state from Electron's internal state somehow.

But then neither of these feel ideal: cached data on the PHP side could quickly fall out of step with reality and extra requests to Electron would create overhead. Worth experimenting.

It feels to me like the [`WindowManager`](https://github.com/NativePHP/laravel/blob/main/src/Windows/WindowManager.php) could be the place to take care of this when the time comes.

**Comment by danielpetrica** [comment]
I recently encountered this problem and had to use a "state manger" and some events to see if the main apps window is open or not. This would be very useful in apps where windows can be opened and closed all the time

**Comment by simonhamp** [comment]
This is [in progress](https://github.com/NativePHP/laravel/pull/391)

**Comment by simonhamp** [comment]
You can [now](https://github.com/NativePHP/laravel/pull/391) use the `Window::current()` and `Window::get()` facade methods to get the current active window or a named window which will return a `Window` object with the state of that window at the time of the API call.
